base/acme/src/main/java/org/dogtagpki/acme/server/ACMERequestFilter.java

```diff
@@ -8,23 +8,25 @@
 import javax.ws.rs.ServiceUnavailableException;
 import javax.ws.rs.container.ContainerRequestContext;
 import javax.ws.rs.container.ContainerRequestFilter;
-import javax.ws.rs.container.PreMatching;
 import javax.ws.rs.ext.Provider;
 
 /**
  * @author Fraser Tweedale
  */
 @Provider
-@PreMatching  // run filter before JAX-RS request matching
```

Comment:
I'm not familiar with this either, but I was just following the example here:
https://stackoverflow.com/questions/23641345/jersey-request-filter-only-on-certain-uri

I've tested the code, the `pki acme-info` will return Available or Unavilable correctly based on `pki acme-enable/disable` invocations.

Reply:
I guess we can't use `@PreMatching` since we still want to match certain operations such as `login`, `logout`, `enable`, and `disable`. If we use `@PreMatching` none of those operation will work if the service is disabled.

Another option is to move those operations to `/pki` instead of `/acme`, but that will require some additional work.

Reply:
Ahhh I see. That makes sense. I'm fine with it as-is.